src/Moq/ActionObserver.cs

```diff
@@ -101,8 +101,9 @@ Expression[] GetArgumentExpressions(Invocation invocation, Match[] matches)
 				{
 					expressions[i] = Expression.Constant(invocation.Arguments[i], parameterTypes[i]);
 
-					// Add a `Convert` node (like the C# compiler does) in places where type coercion happens:
-					if (invocation.Arguments[i] != null && invocation.Arguments[i].GetType() != parameterTypes[i])
+					// Add a `Convert` node (like the C# compiler does) in places where nullable type coercion happens:
```

Comment:
It might be a good idea to add a unit test fixture that documents the C# compiler's behavior with regard to `Convert` nodes, so that we're really sure what should happen here.

Reply:
Done, but I'm still not feeling super-confident about this. The rules for when `Convert` nodes are required are somewhat arcane.